### DISCUSSION TITLE
Metadata fix - How to include error print to log file?

### DISCUSSION THREAD

**Discussion by HannahGaenslen** [ESMValGroup/ESMValCore#2090] [Q&A]
Hi all, 

I had a problem with the metadata of CNRM models when running `recipe_radiation_budget.py`. @mpmoine has explained and asked about the error in this [GitHub discussion](https://github.com/ESMValGroup/ESMValTool/discussions/3160). 
Right now I'm trying to fix the issue with a metadata fix in ESMValCore, but the testing wasn't successful yet. I wanted to print a potential error to the log file, but the logging method didn't work.
```
import logging
logger = logging.getLogger(__name__)
...
logger.info("something")
```
What did I do wrong? Or is there another method to print information from a metadata fix to the log file?

Thanks for your help, 
Hannah 

[recipe_radiation_budget_20230614_072655_main_log_debug.txt](https://github.com/ESMValGroup/ESMValCore/files/11743472/recipe_radiation_budget_20230614_072655_main_log_debug.txt)
[recipe_radiation_budget_20230614_072655.txt](https://github.com/ESMValGroup/ESMValCore/files/11743474/recipe_radiation_budget_20230614_072655.txt)
[cnrm_cm6_1.txt](https://github.com/ESMValGroup/ESMValCore/files/11743476/cnrm_cm6_1.txt)

**Comment by HannahGaenslen** [comment]
As an addition to my question: I'm sure that the issue is not that my modified cnrm_cm6_1.py fix isn't used during the recipe execution. Checking that was possible with a wrong ident in the python function:
```
  File "/data/home/globc/gaenslen/ESMValTool/sources/ESMValCore/esmvalcore/cmor/_fixes/cmip6/cnrm_cm6_1.py", line 111
    logger.info('HANNAH')
                         ^
IndentationError: unindent does not match any outer indentation level
```

**Comment by ehogan** [comment]
Hi @HannahGaenslen 👋 Thank you for getting in touch! 😊 Apolgies for not responding sooner; the last few weeks have been really busy for me! 😩

I can't immediately see anything wrong with your logging code. Other modules in that area of ESMValCore have the same logging code!

The first error in the log you provided (L3016) states:
```
2023-06-14 07:27:20,337 UTC [391591] ERROR   esmvalcore.preprocessor:353 Failed to run preprocessor function 'fix_metadata' on the data
```
This suggests that ESMValTool is correctly finding your `fix_metadata` method on your class, but it can't run it.

Can you simplify the `fix_metadata` method? For example, update the method so that it simply `return cubes`, with no other code. Does the error go away if you do this?

**Reply by HannahGaenslen** [reply]
Hi @ehogan, 
Thank you for your answer! 😊 
I updated it in the way you mentioned it, but the error remains the same. 

[main_log_debug_simple.txt](https://github.com/ESMValGroup/ESMValCore/files/11869488/main_log_debug_simple.txt)

**Reply by ehogan** [reply]
Hmmm 🤔 Are you planning to attend the workshop this week? I'm not familiar with how the fixing code works, so there may be something I'm missing (what you have done seems to align with [ESMValCore documentation: Fixing data](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/develop/fixing_data.html#fixing-data)) 🤔

**Reply by HannahGaenslen** [reply]
No unfortunately not since this week was the last week of my internship at CERFACS. But thank you anyway. :)
